### PR TITLE
Docs: Edit docs in "Publishing" section of site

### PR DESCRIPTION
**Pull request by josmperez**
Edit for style, flow, and updates:
- sign a plugin
- package a plugin
- publish or update a plugin
Notes:
- remove "Publishing and signing criteria" topic in favor of linking to Plugin Policy. Grafana redirects are already in place, no links will be broken. 
- remove "signature levels" in "Sign a plugin" doc in favor of linking to Plugin Policy
- add "Plugin Deprecation Policy" verbage and link to "publish or update a plugin" doc and FAQ. 

Open questions: 
1. Should plugin deprecation policy be its own separate page?
2. In "Update a plugin", how is process different after plugin update is successfully submitted?

### CODE REVIEW

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (marefr) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:23]:
```suggestion
1. [Follow our guidelines](https://grafana.com/legal/plugins/#plugin-publishing-and-signing-criteria) - Learn about the Grafana Labs criteria for publishing and signing plugins. 
```

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (marefr) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:124]: I know this haven't changed, but wonder if we want to push some more for using our catalog and less on the alternative methods.

**Review by marefr — APPROVED** [review]
LGTM great just a note on some unchanged content :)

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (marefr) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:83]: On your question. Given the title of this page is _Publish or update a plugin_ it could make sense to put it in a separate page. However, it's still related to deprecate a plugin so could stay as well. For now, lets keep it here would be my suggestions

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (marefr) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:81]: Regarding your question 

> In "Update a plugin", how is process different after plugin update is successfully submitted?

I think what you've written here make sense. The main difference as you point out is the existing test environment that should speed things up.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:124]: I see what you mean. I've separated the "unlist" question from "deprecated" and moved the latter to later in the FAQ.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:83]: I have at least temporarily moved deprecation to a separate page AND kept it on this page as an FAQ. I'm open to other ideas though if other reviewers feel strongly about one way or the other.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:24]: I think we could probably tidy this section up a little, we don't need the multiple links to signature level, and in fact I think we could omit that entire section as we're better off linking to the full guidance - users will have to read that anyway. I think we also shouldn't link directly to the sign instructions for either type as users may skip the generation of an access policy token

This would be my suggestion, but also happy to discuss more or for alternatives

> Plugins can have different [signature levels](https://grafana.com/legal/plugins/#what-are-the-different-classifications-of-plugins) according to their author, related technology, and intended use. 
>
> The plugin can be _public_, signed as Community or Commercial, and distributed within the [Grafana plugins catalog](https://grafana.com/grafana/plugins/), so that it is available for others to install.  Or _private_ and only available for use within your organization.
>
> Before signing your plugin, review the [Plugins policy](https://grafana.com/legal/plugins/) to determine the appropriate signature for your plugin.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:56]: This currently appears on the

 same line, maybe needs 2 whitespace characters after all stacks?

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:66]: We should maybe have a next steps here to present the choice of the subsequent sections (public/private)?

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md]: @sympatheticmoose - Shoot, I forgot to check in the commit (deprecation.md is now added). I'll leave it up to you to decide if/where it goes up. Could you clarify if you want to move it to create-a-plugin/extend-a-plugin or leave it out? I'm not 100% sure based on your comment above.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md]: that works well.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:22]: okay, let's remove step one and make these bullets.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:23]: done.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:24]: yes, let's add that new point. I would hate to lose the examples link because we don't do a great job right now of publicizing them. I've moved it up to the previous bullet point list.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md]: i've switched it to "a" here and one other place later on.

BTW, this FAQ is a good example of content that I want to eventually raise several questions up into the main section of the topic as topics in themselves. The fewer FAQs the better to improve organization of content.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:20]: done.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md]: That's more concise. Shorter is good. :)

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:64]: As a rule of thumb, technical writers separate out distinct topics for usability and discoverability of information based on each distinct task, even if the differences are minor. In this particular case, however, the differences are very minor and you have suggested a way to organize the content that I hadn't thought of. Shorter is good, I like it.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:24]: done.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:64]: agreed.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:54]: done.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (sympatheticmoose) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:24]: The signature levels section is still there - I still think we should remove it to avoid repetition and the risk of folks missing the access policy step

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (sympatheticmoose) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:68]: I prefer a section and left a suggestion on the content for it here https://github.com/grafana/plugin-tools/pull/462#discussion_r1366816623.

Note the empty page is still showing for me
Screenshot 2023-10-23 at 14 44 22

Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:68]: Should be resolved by https://github.com/grafana/plugin-tools/pull/462/commits/bed1e6c29eb6f316177a96fe9d32ed925d20859d

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:24]: I'm not sure why you were seeing the sign levels section, it was removed. Please check again.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:57]: We state in the plugins policy that we require the source code to be public for community plugins. We can probably remove "If you're unable to make the source code publicly available, let us know in a comment on your plugin submission"

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:59]: We can remove "For more advanced plugins", _may_ is enough wiggle room as really we want plugin developers to help us with the testing as that makes it a far faster process for everyone.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (sympatheticmoose) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:68]: Screenshot 2023-10-26 at 20 45 49
still there

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (sympatheticmoose) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:24]: It's still there
Screenshot 2023-10-26 at 20 49 55

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (sympatheticmoose) [inline docusaurus/docs/publish-a-plugin/sign-a-plugin.md:32]: Commenting again to place it ^ sig levels ection

**Review by sympatheticmoose — COMMENTED** [review]
Getting there :) Still some things to address, but assuming these are done I'd have no objection to merging

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:24]: oh, THAT signature levels section. I thought you were talking about the one identical to https://grafana.com/legal/plugins/#what-are-the-different-classifications-of-plugins which was removed. done, thanks.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:57]: done.

File: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md:59]: done.

File: docusaurus/docs/publish-a-plugin/sign-a-plugin.md
Comment (josmperez) [reply docusaurus/docs/publish-a-plugin/sign-a-plugin.md:32]: It is above where the sig levels section used to be.

**Comment by josmperez** [comment]
I am getting a blocking error when attempting to build the site so I'm going to hold off on merging until I can verify the changes.

**Comment by sympatheticmoose** [comment]
@josmperez I've pushed it to dev if you want to verify https://grafana-dev.com/developers/plugin-tools/publish-a-plugin/publish-a-plugin

Also pulled locally and build ran fine for me, think you're good to go

**Pull request merged**
